lib/dentaku/date_arithmetic.rb

```diff
@@ -47,5 +43,18 @@ def sub(duration)
           "'#{duration || duration.class}' is not coercible for date arithmetic"
       end
     end
+
+    private
+
+    def change_datetime(base, unit, value)
+      case unit
+      when :year
+        base >> (value * 12)
```

Comment:
This way hours, minutes and seconds are also preserved on the original `@base` object.